tests/unit/backend/corpora/api_server/curator/collection/test_curation_collections.py

```diff
@@ -408,17 +411,83 @@ def test__update_collection__OK(self):
+    def test__update_collection_partial_data__OK(self, get_doi: Mock):
+        get_doi.return_value = "stub:doi"
+        links = [{"link_name": "name", "link_type": "RAW_DATA", "link_url": "http://test_link.place"}]
+        name = "partial updates test collection"
+        description = "some description"
+        contact_name = "first last"
+        contact_email = "first@email.top_level_domain"
+
+        # Update the name ONLY
+        new_name = "partial updates 2"
+        collection = self.generate_collection(
+            self.session,
+            name=name,
+            description=description,
+            contact_name=contact_name,
+            contact_email=contact_email,
+            links=links,
+        )
+        collection_id = collection.id
+
+        metadata = {"name": new_name}
         response = self.app.patch(
             f"/curation/v1/collections/{collection_id}",
             data=json.dumps(metadata),
             headers=self.make_owner_header(),
         )
         self.assertEqual(200, response.status_code)
         response = self.app.get(f"curation/v1/collections/{collection_id}")
-        self.assertEqual(response.json["name"], metadata["name"])
+        self.assertEqual(response.json["name"], new_name)
+        self.assertEqual(response.json["description"], description)
+        self.assertEqual(response.json["contact_name"], contact_name)
+        self.assertEqual(response.json["contact_email"], contact_email)
+        self.assertEqual(response.json["links"], links)
+
+    @patch("backend.corpora.common.entities.Collection.get_doi")
+    # Patch Collection.get_doi to ensure the additional collection.update call is made in the update_collection method
+    def test__update_collection__links_management__OK(self, get_doi: Mock):
+        get_doi.return_value = "stub:doi"
+        links = [
+            {"link_name": "name", "link_type": "RAW_DATA", "link_url": "http://test_link.place"},
+            {"link_name": "second link name", "link_type": "RAW_DATA", "link_url": "http://other_test_link.place"},
+        ]
+        new_links = [
+            {"link_name": "new link", "link_type": "RAW_DATA", "link_url": "http://brand_new_link.place"},
+            {"link_name": "new link 2", "link_type": "RAW_DATA", "link_url": "http://brank_new_link_two.place"},
+        ]
+        name = "partial updates test collection"
+
+        links_configurations = (
+            ("With links already in place; new links replace old", links, new_links, 200, new_links),
+            ("With no links in place; new links get added", None, new_links, 200, new_links),
+            ("With links in place, but empty request; links persist", links, None, 200, links),
+            ("With links in place, empty array passed; BAD REQUEST 400", links, [], 400, links),
+        )
+
+        for links_config in links_configurations:
```

Comment:
unpack the link_config here rather than bellow.

Reply:
Yikes, great point. will change. 🤦‍♂️